src/main/java/nl/hsac/fitnesse/fixture/util/HttpClient.java

```diff
@@ -142,6 +147,20 @@ protected void getResponse(String url, HttpResponse response, HttpRequestBase me
             for (Header h : resp.getAllHeaders()) {
                 responseHeaders.put(h.getName(), h.getValue());
             }
+        	if(responseHeaders.containsKey("Set-Cookie"))
+            {
+            	StringBuilder cookieHeader = new StringBuilder();
+            	List<Cookie> cookies = store.getCookies();
+        		
+        		for(Cookie cookie : cookies)
+        		{
+        			cookieHeader.append(cookie.getName() + "=" + cookie.getValue());
+        			cookieHeader.append(";");
```

Comment:
As far as I can tell header values should be separated by `,` not `;`. 

Still all cookies are being added to the headers (also any that were present in the store before the request). We should find a way to have only the ones from the response.